### PR TITLE
Add strict flag check for ecma_op_object_put_apply_receiver

### LINKED ISSUE #4441: Assertion 'ecma_is_value_true (result)' failed in ecma_regexp_split_helper
###### JerryScript revision
fdaacde

###### Build platform
Ubuntu 18.04.5 LTS(Linux 4.15.0-119-generic x86_64)

###### Build steps
```
./tools/build.py --clean --debug --compile-flag=-fsanitize=address \
--compile-flag=-m32 --compile-flag=-fno-omit-frame-pointer \
--compile-flag=-fno-common --compile-flag=-g --strip=off \
--system-allocator=on --logging=on --linker-flag=-fuse-ld=gold \
--error-messages=on --profile=es2015-subset  
```

###### Test case
```
var get = [];
var constructor = Function();
constructor[Symbol.species] = Object;
var p = new Proxy({ constructor: constructor, flags: '', exec: function() { return /\ua796/iu; } }, { defineProperty: function(o, k) { get.push(k); return o[k]; }});
RegExp.prototype[Symbol.split].call(p, 7996);

```

###### Output
```
ICE: Assertion 'ecma_is_value_true (result)' failed at /root/jerryscript/jerry-core/ecma/operations/ecma-regexp-object.c(ecma_regexp_split_helper):2277.
Error: ERR_FAILED_INTERNAL_ASSERTION
Aborted

```
Credits: Found by chong from OWL337.

### PR DESCRIPTION
**Pull request by rerobika**
This patch fixes #4441.

JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-objects.c [hunk 1]
```diff
@@ -1164,6 +1164,11 @@ ecma_op_object_put_apply_receiver (ecma_value_t receiver, /**< receiver */
 
       /* 5.e.iv */
       result = ecma_op_object_define_own_property (receiver_obj_p, property_name_p, &prop_desc);
+
+      if (JERRY_UNLIKELY (ecma_is_value_false (result)) && is_throw)
+      {
+        result = ecma_raise_type_error (ECMA_ERR_MSG ("Proxy trap returned falsish"));
```
Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-objects.c:1170]: Proxy put trap returned with false.

**Review by zherczeg — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/operations/ecma-objects.c (hunk 1 above)
Comment (rerobika) [reply jerry-core/ecma/operations/ecma-objects.c:1170]: That's a matter of taste. Your suggestion is equivant to the SpiderMonkey's error message mine is to V8. To keep the consistecy, I'd leave the V8 error message.

Comment (zherczeg) [reply jerry-core/ecma/operations/ecma-objects.c:1170]: Consistency?

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**